### ISSUE TITLE
Map Server abruptly and mercilessly aborts | use_sql_db: yes

### ISSUE THREAD

**Issue by SpineOfSteel** [rathena/rathena#7925]
### rAthena Hash

6fb5120b8

### Client Date

2023-09-08

### Server Mode

Renewal

### Result

- rAthena compiles with No errors, No warnings
- I fixed db/re/mob_db.yml LINT issue. Still no luck. I don't think its DB/YML issue
- Added #define PACKETVER 20200401 @ "src\custom\defines_pre.hpp" (I guess error is due to PACKETVER conflicts) [[Step 10 -- rAthena: Installing, Configuring, and Optimizing  -- Check how to reproduce ](https://forums.openkore.com/viewtopic.php?t=333763)]
- Strangely I don't see any Debug/Error Info.. even in the log folder. Map-server just aborts instantly!

![image](https://github.com/rathena/rathena/assets/14861013/5b61666d-a8d8-4d75-9873-1b8d12a9ec46)

### Expected Result

if Map Server works as intended, it would be great

### How to Reproduce

[[[ rAthena: Installing, Configuring, and Optimizing ] Step. 01 to 30](https://forums.openkore.com/viewtopic.php?t=333763)]

### Official Information

Localhost Only , Client kRO_FullClient_20210406

**Comment by SpineOfSteel** [comment]
Incompatible SQL file seems to be the culprit. Replaced sql-files/mob_db_re.sql with older [mob_db_re.sql ](https://github.com/rathena/rathena/blob/6ccf15330e181ee4ce3bb061a52e1c2667b182a5/sql-files/mob_db_re.sql)(Dec 2021). The Map server doesn't crash and I can see errors and warnings in the logs (PFA)

[map-msg_log.log](https://github.com/rathena/rathena/files/12568233/map-msg_log.log)

**Comment by reunite-ro** [comment]
From you error above, have this lines

(09/10/2023 05:13:45) [ SQL Error ] : DB error - Unknown column 'racegroup_malangdo' in 'field list'

Are you sure you already have this column in mob_db_re ?

Didnt tried your rathena hash yet.

**Comment by SpineOfSteel** [comment]
I was checking diff between old and new mob_db_re schema..  Found 4 new columns and 1 more column racegroup_illusion_turtle 

I will experiment a bit more and keep you posted
![image](https://github.com/rathena/rathena/assets/14861013/687efead-e137-4f5d-88a4-4f288fd9a927)

**Comment by SpineOfSteel** [comment]
Just to confirm adding the abovementioned columns does crash the map server (without these it works). In such cases.. Instead of Aborting A warning must suffice I guess.. 

![image](https://github.com/rathena/rathena/assets/14861013/9fc6c218-99a2-4674-b817-29af03a6eabb)

**Comment by reunite-ro** [comment]
Did u git clone and run all the sql ? And are u sure u have mob_db_re in your database ? Also appears 'null' from your screenshot which is should be mob_db from my following screenshot.

Mine renewal looks like this.
![image](https://github.com/rathena/rathena/assets/11705163/5aed790d-139b-46a9-aa70-ef1de6037da8)



Ive tried latest git for today, https://github.com/rathena/rathena/commit/6fb5120b8c5e102b37518c4d275db365c36e689a but seems like no issue about that.

**Comment by SpineOfSteel** [comment]
yep yep yep. git clone > Convert all data files to SQL -- yaml2sql > Executed all SQL files under rathena\sql-files

Will try once more without yaml2sql

**Comment by reunite-ro** [comment]
> yep yep yep. git clone > **Convert all data files to SQL -- yaml2sql** > Executed all SQL files under rathena\sql-files
> 
> Will try once more without yaml2sql

May I know why did you converting to sql ? you can just run the sql in sql-files folder.

**Comment by SpineOfSteel** [comment]
I thought about it before... If you observe yaml2sql, it does detect 4/5 files that can be converted to SQL. 
I think you are absolutely right, this conversion is unnecessary and redundant if SQL files are up-to-date

![image](https://github.com/rathena/rathena/assets/14861013/093a64a3-f543-4d18-90e8-94cc641c5df2)

**Comment by vstumpf** [comment]
Can you run the map server in visual studio? Running it in a debugger will tell you where exactly the crash is.

**Comment by SpineOfSteel** [comment]
I was experimenting with conf's 
rathena\conf\import\inter_conf.txt 
`use_sql_db: yes`
![image](https://github.com/rathena/rathena/assets/14861013/bda97a25-9d27-4845-8c42-c98f3136fe99)


I think its 
Works as expected with 
`use_sql_db: no`

![image](https://github.com/rathena/rathena/assets/14861013/4c5c7983-3e5a-4296-89b6-9339efe93a9b)

**Comment by SpineOfSteel** [comment]
Stack:
- MariaDB Server 10.9.8 
- PHP 7.4.28
- HeidiSQL 12.5
- Nginx 1.25.2
- Laragon Portable Mint 5.0.0

If I drop these columns on my SQLDB, it doesn't crash. @vstumpf  Sure Will try to debug
![image](https://github.com/rathena/rathena/assets/14861013/1b29ab42-7c8c-4180-9fbd-3572c4fdbb14)

**Comment by vstumpf** [comment]
The sql files in `sql` are not updated to the latest rathena. If you want to use sql, you have to use the `yaml2sql` converter.

 The racegroups are supposed to match (in order) the ones here: https://github.com/rathena/rathena/blob/6fb5120b8c5e102b37518c4d275db365c36e689a/src/map/map.hpp#L361-L367

If the sql table is missing one that the map-server expects, it will probably crash. It needs to match.

**Comment by SpineOfSteel** [comment]
Here, map server trying to convert string to int (stoi) on "element" column.. which is varchar.

- Clearly, during mob_read_sqldb_sub (i.e Convert SQL data to YAML Node) str array is offset by one or two
- At position  i= 32 and index = 24, 
- so str[i + index] is basically str[56] which is string. Check line mob.cpp @line 5124 . Can't convert string to int!

![image](https://github.com/rathena/rathena/assets/14861013/5b5df897-3a35-4f68-9a96-954ed0901783)

**Comment by SpineOfSteel** [comment]
TLDR; 
Well, Temporarily i set `use_sql_db: no` in `athena\conf\import\inter_conf.txt` config file and it works as intended. Config with `use_sql_db: yes` is glitchy. Too many bugs to resolve even if mob_db_re issues are temporarily resolved. If anyone here was able to successfully run with `use_sql_db: yes`, Please let me know.. references are welcome!

As per map.hpp, I tried to... 
- add missing 'racegroup_illusion_turtle' in SQL query [here in mob.cpp Ln 5280](https://github.com/rathena/rathena/blob/6fb5120b8c5e102b37518c4d275db365c36e689a/src/map/mob.cpp#L5280)
- add missing 'racegroup_illusion_turtle' column in mob_db_re table
- No changes made in map.hpp
![image](https://github.com/rathena/rathena/assets/14861013/23a981f8-7ce1-49a8-b558-f82add08ae4b)

Map server didn't crash but there are lots of errors!  (PFA log)

[map-msg_log.log](https://github.com/rathena/rathena/files/12574109/map-msg_log.log)

**Comment by SpineOfSteel** [comment]
#6323

**Comment by SpineOfSteel** [comment]
Thanks for the updates and swift response! Issues logged in map-msg_log not related to this thread.
